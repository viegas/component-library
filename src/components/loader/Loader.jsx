import React from 'react';
import PropTypes from 'prop-types';

import * as Styled from './Loader.styles';

export const config = {
    sm: {
        size: 20,
        strokeWidth: 1,
    },
    md: {
        size: 30,
        strokeWidth: 2,
    },
    lg: {
        size: 50,
        strokeWidth: 3,
    },
    xl: {
        size: 80,
        strokeWidth: 4,
    },
    xxl: {
        size: 100,
        strokeWidth: 6,
    },
};

const Loader = ({ color, size }) => {
    const options = config[size];

    return <Styled.Spinner {...options} color={color} />;
};

Loader.propTypes = {
    size: PropTypes.oneOf(['sm', 'md', 'lg', 'xl', 'xxl']),
    color: PropTypes.string,
};

Loader.defaultProps = {
    size: 'sm',
    color: 'blue',
};
export default Loader;
