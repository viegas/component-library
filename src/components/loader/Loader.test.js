import Loader from './Loader';
import * as Styled from './Loader.styles';

describe('<Loader />', () => {
    it('Should render correctly', () => {
        const wrapper = shallow(<Loader />);
        expect(wrapper).toMatchSnapshot();
    });

    styledComponentsSnapshots(Styled, '*');
});
