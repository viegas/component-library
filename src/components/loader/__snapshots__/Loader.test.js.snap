// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> <Styled.Spinner /> Should render correctly 1`] = `
.c0 {
  height: px;
  width: px;
  border-radius: 50%;
  border: px solid rgba(0,0,0,0.1);
  -webkit-animation: ibwhYF 800ms ease infinite;
  animation: ibwhYF 800ms ease infinite;
}

<div
  className="c0"
/>
`;

exports[`<Loader /> Should render correctly 1`] = `
<Loaderstyles__Spinner
  color="#f96d00"
  size={20}
  strokeWidth={1}
/>
`;
